distage/distage-model/src/main/scala/izumi/distage/model/definition/dsl/ModuleDefDSL.scala

```diff
@@ -149,6 +153,71 @@ object ModuleDefDSL {
     }
   }
 
+  final class MultipleDSL[I]
+  (
+    protected val mutableState: MultipleRef
+    , protected val key: DIKey.TypeKey
+  ) extends MultipleDSLMutBase[I] {
+
+    def named[T](name: T)(implicit idContract: IdContract[T]): MultipleNamedDSL[I] = {
+      addOp(MultipleInstruction.SetId(name))(new MultipleNamedDSL[I](_, key.named(name)))
+    }
+
+    override def to[T >: I : Tag]: MultipleBindDSL[I] = {
+      addOp(ImplWithReference(DIKey.get[T]))(ref => new MultipleBindDSL[I](ref, key))
+    }
+
+    override def to[T >: I : Tag](name: String): MultipleBindDSL[I] = {
+      addOp(ImplWithReference(DIKey.get[T].named(name)))(ref => new MultipleBindDSL[I](ref, key))
+    }
+
+    def tagged(tags: BindingTag*): MultipleDSL[I] =
+      addOp(MultipleInstruction.AddTags(tags.toSet)) {
+        new MultipleDSL[I](_, key)
+      }
+  }
+
+  final class MultipleBindDSL[I]
+  (
+    protected val mutableState: MultipleRef
+    , protected val key: DIKey.TypeKey
+  ) extends MultipleDSLMutBase[I] {
+
+    override def to[T >: I : Tag]: MultipleBindDSL[I] = {
+      addOp(ImplWithReference(DIKey.get[T]))(ref => new MultipleBindDSL[I](ref, key))
+    }
+
+    override def to[T >: I : Tag](name: String): MultipleBindDSL[I] = {
+      addOp(ImplWithReference(DIKey.get[T].named(name)))(ref => new MultipleBindDSL[I](ref, key))
+    }
+  }
+
+  final class MultipleNamedDSL[I]
+  (
+    protected val mutableState: MultipleRef
+    , protected val key: DIKey.IdKey[_]
+  ) extends MultipleDSLMutBase[I] {
+
+    override def to[T >: I : Tag]: MultipleNamedDSL[I] = {
```

Comment:
You should be able to use `tagged` after `named`, but NOT after `to`. Could you allow tagged here, make `to` go to BindDSL here too (to impl should be the same for all of them now, so you may extract it into upper trait) and add a test that `bind[X].named(x).tagged(y)` is ok?

Reply:
Done.